From a92d4ca460f76b22ee7d68d50a9a99ee7c8f6f5c Mon Sep 17 00:00:00 2001
From: Michael Dawson <mdawson@devrus.com>
Date: Mon, 10 Jul 2017 15:17:58 -0400
Subject: [PATCH] deps: Debug code requires bigger buffer on s390

s390 requires a bigger buffer for debug code
when snapshots are not enabled.

PR-URL: https://github.com/nodejs/node-private/pull/93
Reviewed-By: Myles Borins <Myles.Borins@gmail.com>

Patch-Source: https://github.com/nodejs/node/commit/a92d4ca460
---
 deps/v8/include/v8-version.h | 2 +-
 deps/v8/src/builtins.cc      | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/deps/v8/include/v8-version.h b/deps/v8/include/v8-version.h
index 2162538fda..55d32cc899 100644
--- a/deps/v8/include/v8-version.h
+++ b/deps/v8/include/v8-version.h
@@ -11,7 +11,7 @@
 #define V8_MAJOR_VERSION 5
 #define V8_MINOR_VERSION 1
 #define V8_BUILD_NUMBER 281
-#define V8_PATCH_LEVEL 101
+#define V8_PATCH_LEVEL 102
 
 // Use 1 for candidates and 0 otherwise.
 // (Boolean macro values are not supported by all preprocessors.)
diff --git a/deps/v8/src/builtins.cc b/deps/v8/src/builtins.cc
index 01f103101c..4d399b2ca0 100644
--- a/deps/v8/src/builtins.cc
+++ b/deps/v8/src/builtins.cc
@@ -4720,8 +4720,8 @@ Handle<Code> MacroAssemblerBuilder(Isolate* isolate,
 #ifdef DEBUG
   // We can generate a lot of debug code on Arm64.
   const size_t buffer_size = 32 * KB;
-#elif V8_TARGET_ARCH_PPC64
-  // 8 KB is insufficient on PPC64 when FLAG_debug_code is on.
+#elif V8_TARGET_ARCH_PPC64 || V8_TARGET_ARCH_S390X
+  // 8 KB is insufficient on PPC64 and S390X when FLAG_debug_code is on.
   const size_t buffer_size = 10 * KB;
 #else
   const size_t buffer_size = 8 * KB;
